Code/autopkglib/URLDownloader.py

```diff
@@ -116,45 +104,99 @@ class URLDownloader(Processor):
         },
     }
 
-    def main(self):
-        if not is_mac():
-            raise ProcessorError("This processor is Mac-only!")
-        # clear any pre-exising summary result
+    def getxattr(self, attr):
+        """Get a named xattr from a file. Return None if not present"""
+        if attr in xattr.listxattr(self.env["pathname"]):
+            return xattr.getxattr(self.env["pathname"], attr)
+        return None
+
+    def prepare_curl_cmd(self, pathname_temporary):
```

Comment:
Method split from `main()`. Copied [lines](https://github.com/autopkg/autopkg/blob/15c6307254d4ef1d1dbb8a2648a204a340f5b967/Code/autopkglib/URLDownloader.py#L168) 168..183 and 194..208. Lines 185..192 moved to URLGetter.